### DISCUSSION TITLE
Audio buffer from the Frame

### DISCUSSION THREAD

**Discussion by crisffreire** [asticode/go-astiav#86] [Q&A] [answered]
Would you have any example of how to get the audio buffer from the Frame?

**Comment by asticode** [comment]
For now there's none for the simple fact that retrieving an audio buffer from a frame has not yet been implemented 🫣 

But I'll take a look this week and keep you posted 👍 

And to make sure we're on the same page, with audio with several channels, you'll end up with one big []byte which will be the concatenation of buffer from channel 1 then buffer from channel 2, etc.

**Reply by asticode** [reply, marked as answer]
FYI I've added methods to manipulate a frame's audio buffer. You can either use [SamplesCopyToBuffer](https://github.com/asticode/go-astiav/blob/master/frame.go#L125) directly or use `Frame.Data()` as described in the [example](https://github.com/asticode/go-astiav/blob/master/examples/frame_data_manipulation/main.go#L27)

Let me know whether that answer your question 👍

**Reply by crisffreire** [reply]
Excellent solution! Thank you very much!👍
